src/Lucene.Net/Search/FieldCacheImpl.cs

```diff
@@ -829,6 +836,20 @@ protected internal override StringIndex CreateValue(IndexReader reader, Entry en
                 StringIndex value_Renamed = new StringIndex(retArray, retArrayOrdered, mterms);
                 return value_Renamed;
             }
+
+            public void Dispose()
```

Comment:
This won't be called from finalizer, no?

Reply:
No, the finalizer is called for the `Segment`